src/components/MenuItem.tsx

```diff
@@ -0,0 +1,599 @@
+                                {Boolean(label) && (
+                                    <View style={icon ? styles.mb2 : null}>
+                                        <Text style={StyleUtils.combineStyles([styles.sidebarLinkText, styles.optionAlternateText, styles.textLabelSupporting, styles.pre])}>{label}</Text>
+                                    </View>
+                                )}
+                                <View style={[styles.flexRow, styles.pointerEventsAuto, disabled && styles.cursorDisabled]}>
+                                    {Boolean(icon) && Array.isArray(icon) && (
+                                        <MultipleAvatars
+                                            isHovered={isHovered}
+                                            isPressed={Boolean(pressed)}
+                                            icons={icon}
+                                            size={avatarSize}
+                                            secondAvatarStyle={[
+                                                StyleUtils.getBackgroundAndBorderStyle(theme.sidebar),
+                                                pressed && interactive ? StyleUtils.getBackgroundAndBorderStyle(theme.buttonPressedBG) : undefined,
+                                                isHovered && !pressed && interactive ? StyleUtils.getBackgroundAndBorderStyle(theme.border) : undefined,
+                                            ]}
+                                        />
+                                    )}
+                                    {icon && !Array.isArray(icon) && (
+                                        <View style={[styles.popoverMenuIcon, iconStyles, StyleUtils.getAvatarWidthStyle(avatarSize)]}>
+                                            {typeof icon !== 'string' && iconType === CONST.ICON_TYPE_ICON && (
+                                                <Icon
+                                                    hovered={isHovered}
+                                                    pressed={pressed}
+                                                    src={icon}
+                                                    width={iconWidth}
+                                                    height={iconHeight}
+                                                    fill={iconFill ?? StyleUtils.getIconFillColor(getButtonState(focused || isHovered, pressed, success, disabled, interactive), true)}
+                                                />
+                                            )}
+                                            {icon && iconType === CONST.ICON_TYPE_WORKSPACE && (
+                                                <Avatar
+                                                    imageStyles={[styles.alignSelfCenter]}
+                                                    size={CONST.AVATAR_SIZE.DEFAULT}
+                                                    source={icon as AvatarSource}
```

Comment:
This one I had to map `as AvatarSource` because TypeScript still thought that `icon` can be of `FC<SvgProps>` type - although I used discriminative union type while declaring `icon` at the top of this file. So not sure if this can be done without `as` @blazejkustra